### PR TITLE
Facebook sdk

### PR DESCRIPTION
**Pull request by darkracer**
-added the facebook sdk and configuered it.
-added a login button, but that only displays logout messages.

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,21 +1,24 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     package="assembly.giraff" >
+    <uses-permission android:name="android.permission.internet"/>
```
Comment (intrications) [inline app/src/main/AndroidManifest.xml]: This should be:

<uses-permission android:name="android.permission.INTERNET"/>

**Comment by intrications** [comment]
Looks good to me apart from three issues:

1. The Internet permission mentioned above.

2. If I log in, log out then try to log in again I get an no Android key hashes error. Update: Actually it just doesn't allow me to log in again even after uninstalling and reinstalling the app.

3. The facebook authorization page says "choosefun" which is confusing when the name of the app is "Fun" - this presumably is an issue with the settings on Facebook not within the app.

**Comment by darkracer** [comment]
I think the login problem is not related to SDK settings but to the button handling it.
This is wierd because the button is straight from the tutorial website on facebook.

**Comment by darkracer** [comment]
The login problem is probably something with hashes. I have never entered a hash anywhere so I think
a setting on the app's page is not right.

**Comment by darkracer** [comment]
I think I've found the issue.
https://developers.facebook.com/docs/android/getting-started
When you scroll down to creating developer key hashes you can see why it is not working.
When you follow those steps it should hopefully work.
Let me know if something I did is preventing you from doing this so I can help you.

**Comment by darkracer** [comment]
I got it working with the right app. I see I have created a merge conflict and will try to resolve that in a minute.

**Comment by adrianonco** [comment]
Great @darkracer, just push the PR when finish in the bounty please. Thanks!

**Comment by darkracer** [comment]
@adrianonco I fixed the merge conflict and that finishes the bounty.

**Comment by seuros** [comment]
Could you rebase/squash commits ?

**Comment by darkracer** [comment]
I can't get is to work. I have squashed them and it says so locally but when I pushed it just added the single one.

**Comment by intrications** [comment]
@darkracer You probably need to force push. Or failing that, just create a new pull request.

**Comment by seuros** [comment]
You have to force push after squashing.

check your git log.

**Comment by intrications** [comment]
@darkracer You also need to learn to use branches if you work on multiple pull requests at once. All your commits here are also being pushed to https://github.com/asm-products/giraff-android/pull/6/commits because you are working on the master branch for both.

**Comment by darkracer** [comment]
Creating a new one

**Pull request closed without merging**